execution_engine/src/core/engine_state/transfer.rs

```diff
@@ -255,6 +255,28 @@ impl TransferRuntimeArgsBuilder {
                     None => Err(Error::Exec(ExecError::Revert(ApiError::Transfer))),
                 }
             }
+            Some(cl_value) if *cl_value.cl_type() == CLType::PublicKey => {
+                let public_key: PublicKey = match cl_value.clone().into_t() {
+                    Ok(public_key) => public_key,
+                    Err(error) => {
+                        return Err(Error::Exec(ExecError::Revert(error.into())));
+                    }
+                };
+
+                let account_hash = AccountHash::from(&public_key);
+
+                self.to = Some(account_hash.to_owned());
+                match tracking_copy
+                    .borrow_mut()
+                    .read_account(correlation_id, account_hash)
+                {
+                    Ok(account) => Ok(TransferTargetMode::PurseExists(
+                        account.main_purse().with_access_rights(AccessRights::ADD),
+                    )),
+                    Err(_) => Ok(TransferTargetMode::CreateAccount(account_hash)),
+                }
+            }
+
```

Comment:
There's a lot of duplication in this method.  Can we remove it by doing something like this:
```rust
        let imputed_runtime_args = &self.inner;
        let cl_value = imputed_runtime_args
            .get(mint::ARG_TARGET)
            .cloned()
            .ok_or(Error::Exec(ExecError::Revert(ApiError::MissingArgument)))?;

        let mut resolve = |account_hash: AccountHash| -> Result<TransferTargetMode, Error> {
            self.to = Some(account_hash);
            match tracking_copy
                .borrow_mut()
                .read_account(correlation_id, account_hash)
            {
                Ok(account) => Ok(TransferTargetMode::PurseExists(
                    account.main_purse().with_access_rights(AccessRights::ADD),
                )),
                Err(_) => Ok(TransferTargetMode::CreateAccount(account_hash)),
            }
        };

        match *cl_value.cl_type() {
            CLType::URef => {
                let uref: URef = cl_value
                    .into_t()
                    .map_err(|error| Error::Exec(ExecError::Revert(error.into())))?;
                if !self.purse_exists(uref, correlation_id, tracking_copy) {
                    return Err(Error::Exec(ExecError::Revert(ApiError::InvalidPurse)));
                }
                Ok(TransferTargetMode::PurseExists(uref))
            }
            CLType::ByteArray(32) => {
                let account_hash: AccountHash = cl_value
                    .into_t()
                    .map_err(|error| Error::Exec(ExecError::Revert(error.into())))?;
                resolve(account_hash)
            }

            CLType::Key => {
                let account_key: Key = cl_value
                    .into_t()
                    .map_err(|error| Error::Exec(ExecError::Revert(error.into())))?;
                match account_key.into_account() {
                    Some(account_hash) => resolve(account_hash),
                    None => Err(Error::Exec(ExecError::Revert(ApiError::Transfer))),
                }
            }
            CLType::PublicKey => {
                let public_key: PublicKey = cl_value
                    .into_t()
                    .map_err(|error| Error::Exec(ExecError::Revert(error.into())))?;
                resolve(AccountHash::from(&public_key))
            }
            _ => Err(Error::Exec(ExecError::Revert(ApiError::InvalidArgument))),
        }
```